### PR TITLE
IMapRepository and MapRepository added

### PR DESCRIPTION
**Pull request by vasiliv**
IMapRepository and MapRepository added

### CODE REVIEW

File: src/Application/Interfaces/IMapRepository.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.Interfaces
+{
+    public interface IMapRepository : IGenericRepository<Map>
+    {
+
+    }
```
Comment (DoniyorJN) [inline src/Application/Interfaces/IMapRepository.cs:13]: please read about yagni principle.
empty PR...

Comment (vasiliv) [reply src/Application/Interfaces/IMapRepository.cs:13]: Understand

File: src/Application/Interfaces/IGenericRepository.cs [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-    public interface IGenericRepository<T> where T : class
+    public interface IGenericRepository<T>: IDisposable where T : class
     {
-        T GetById(int id);
-        IEnumerable<T> GetAll();        
-        void Add(T entity);        
-        void Remove(T entity);        
+        Task Add(T entity);
+        Task<List<T>> GetAll();
+        Task<T> GetById(Guid id);
```
Comment (DoniyorJN) [inline src/Application/Interfaces/IGenericRepository.cs:15]: please add  async for all async methods -> AddAsync() ..etc

File: src/API/Program.cs [hunk 3]
```diff
@@ -15,6 +15,11 @@
 #region Repositories
 builder.Services.AddTransient(typeof(IGenericRepository<>), typeof(GenericRepository<>));
 builder.Services.AddTransient<IAppUserRepository, AppUserRepository>();
+builder.Services.AddTransient<IMapRepository, MapRepository>();
```
Comment (DoniyorJN) [inline src/API/Program.cs:18]: please register in UOW

File: src/Application/Interfaces/IAppUserRepository.cs [hunk 4]
```diff
@@ -8,6 +8,11 @@
 namespace Application.Interfaces
 {
     public interface IAppUserRepository : IGenericRepository<AppUser>
-    {
+    {   
+        // for override
+        new Task<List<AppUser>> GetAll();
```
Comment (DoniyorJN) [inline src/Application/Interfaces/IAppUserRepository.cs:13]: why do we need a new keyword?

Comment (DoniyorJN) [reply src/Application/Interfaces/IAppUserRepository.cs:13]: if you want to override use override key

File: src/Application/Interfaces/IBookingRepository.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.Interfaces
+{
+    public interface IBookingRepository : IGenericRepository<Booking>
+    {
+    }
```
Comment (DoniyorJN) [inline src/Application/Interfaces/IBookingRepository.cs:12]: YAGNI !

File: src/Application/Interfaces/IMapRepository.cs (hunk 1 above)
Comment (DoniyorJN) [reply src/Application/Interfaces/IMapRepository.cs:13]: YAGNI

File: src/Infrastructure/Repositories/AppUserRepository.cs [hunk 6]
```diff
@@ -11,9 +12,30 @@ namespace Infrastructure.Repositories
 {
     public class AppUserRepository : GenericRepository<AppUser>, IAppUserRepository
     {
-        public AppUserRepository(ApplicationDbContext context) : base(context)
+        public AppUserRepository(ApplicationDbContext context) : base(context) { }
+
+        public override async Task<List<AppUser>> GetAll()
```
Comment (DoniyorJN) [inline src/Infrastructure/Repositories/AppUserRepository.cs:17]: Add async for all async methods

**Review by DoniyorJN — CHANGES_REQUESTED** [review]

File: src/API/Program.cs (hunk 3 above)
Comment (vasiliv) [reply src/API/Program.cs:18]: Done

File: src/Application/Interfaces/IAppUserRepository.cs (hunk 4 above)
Comment (vasiliv) [reply src/Application/Interfaces/IAppUserRepository.cs:13]: Done

File: src/Application/Interfaces/IBookingRepository.cs (hunk 5 above)
Comment (vasiliv) [reply src/Application/Interfaces/IBookingRepository.cs:12]: we will implement it later

File: src/Application/Interfaces/IGenericRepository.cs (hunk 2 above)
Comment (vasiliv) [reply src/Application/Interfaces/IGenericRepository.cs:15]: added async in implememntation

File: src/Infrastructure/Repositories/AppUserRepository.cs (hunk 6 above)
Comment (vasiliv) [reply src/Infrastructure/Repositories/AppUserRepository.cs:17]: added

**Pull request closed without merging**